crypto/src/bls12377/mod.rs

```diff
@@ -539,20 +540,29 @@ impl AggregateAuthenticator for BLS12377AggregateSignature {
 /// Implement VerifyingKeyBytes
 ///
 
-impl TryInto<BLS12377PublicKey> for BLS12377PublicKeyBytes {
+impl TryFrom<BLS12377PublicKeyBytes> for BLS12377PublicKey {
     type Error = signature::Error;
 
-    fn try_into(self) -> Result<BLS12377PublicKey, Self::Error> {
-        // TODO(https://github.com/MystenLabs/sui/issues/101): Do better key validation
-        // to ensure the bytes represent a poin on the curve.
-        BLS12377PublicKey::from_bytes(self.as_ref()).map_err(|_| Self::Error::new())
+    fn try_from(bytes: BLS12377PublicKeyBytes) -> Result<BLS12377PublicKey, Self::Error> {
+        BLS12377PublicKey::from_bytes(bytes.as_ref()).map_err(|_| Self::Error::new())
     }
 }
 
-impl From<BLS12377PublicKey> for BLS12377PublicKeyBytes {
-    fn from(pk: BLS12377PublicKey) -> BLS12377PublicKeyBytes {
-        let mut bytes = [0u8; CELO_BLS_PUBLIC_KEY_LENGTH];
-        pk.pubkey.serialize(&mut bytes[..]).unwrap();
-        BLS12377PublicKeyBytes::new(bytes)
+impl From<&BLS12377PublicKey> for BLS12377PublicKeyBytes {
+    fn from(pk: &BLS12377PublicKey) -> Self {
+        let pk_bytes =
+            pk
+            .bytes
+            .get_or_try_init::<_, eyre::Report>(|| {
+                let mut bytes = [0u8; CELO_BLS_PUBLIC_KEY_LENGTH];
+                pk.pubkey
+                    .as_ref()
+                    .into_affine()
+                    .serialize(&mut bytes[..])
+                    .unwrap();
+                Ok(bytes)
+            })
```

Comment:
The initialisation here seems to be the same we do on other parts as well (ex [here](https://github.com/MystenLabs/narwhal/pull/507/files#diff-ff9b43e2e185ece3a384a642ac5ba4a67e9b37f000eb819dc82c0a752b3fa23fR174) ) . I would recommend either:
1. Extracting in a method under the `BLS12377PublicKey` 
2. Initialise the `bytes` (if not vastly costly) the same time when we do set the `pubkey`

having duplicate code might lead to diverge implementation in the future

Reply:
You can write this as 
```rust
fn from(pk: &BLS12377PublicKey) -> Self {
            BLS12377PublicKeyBytes::new(pk.as_ref().clone())
}
```